test/mock/grpc/mock_management_plane_grpc_server.go

```diff
@@ -44,9 +51,23 @@ func NewManagementGrpcServer() *ManagementGrpcServer {
 		if mgs.connectionRequest == nil {
 			c.JSON(http.StatusNotFound, nil)
 		} else {
+			mgs.connectionMutex.Lock()
 			c.JSON(http.StatusOK, gin.H{
 				"connectionRequest": mgs.connectionRequest,
 			})
+			mgs.connectionMutex.Unlock()
+		}
```

Comment:
If we are gonna lock for responses, should we also lock for the first branch of the if-clause?
```suggestion
        mgs.connectionMutex.Lock()
		if mgs.connectionRequest == nil {
			c.JSON(http.StatusNotFound, nil)
		} else {
			c.JSON(http.StatusOK, gin.H{
				"connectionRequest": mgs.connectionRequest,
			})
		}
		mgs.connectionMutex.Unlock()
```

Reply:
Now that I'm better caffeinated, I realize you're locking for the read of `mgs.updateDataPlaneStatusRequest` but the suggestion still stands, as the if-clause does read that field.

Reply:
good spot. updating that there now